librz/asm/arch/tms320/tms320_dasm.c

```diff
@@ -10,11 +10,32 @@
 #include <sdb.h>
 
 /* private headers */
-#include "tms320_p.h"
 #include "tms320_dasm.h"
 
 #include "c55x_plus/c55plus.h"
 
+#ifndef get_bits
+#define get_bits(av, af, an) (((av) >> (af)) & ((2 << (an - 1)) - 1))
+#endif
+
+static inline ut16 be16(ut16 v) {
+	ut16 value = v;
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+	ut8 *pv = (void *)&v;
+	value = (pv[0] << 8) | pv[1];
+#endif
+	return value;
+}
+
+static inline ut32 be24(ut32 v) {
+	ut32 value = v;
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+	ut8 *pv = (void *)&v;
+	value = (pv[0] << 16) | (pv[1] << 8) | pv[2];
+#endif
+	return value;
+}
```

Comment:
are these still needed?

Reply:
Yes, since they convert `ut32` that is read from the bitfield. Using just the `rz_read_*()` API would require more significant refactoring. Thus I stopped on this middle-ground solution.

Reply:
Actually... I'm not sure what you mean. Isn't `be24` the same thing as `rz_read_be24`?

Reply:
It isn't - just see the input values. `rz_read_be24()` reads memory, this `be24` *converts* the host-endianned *integer*, which is in turn was read from the bitfield with `field_value()` macros. Of course it can be converted to read corresponding memory directly, but its *a lot of work*.

Reply:
I'd need to think a bit more about this... Anyway, tms320.... Ok :D